### DISCUSSION TITLE
Windows 11 - minimizing the server application results in “underwater sounds” for the connected clients

### DISCUSSION THREAD

**Discussion by Willem119** [jamulussoftware/jamulus#2233] [Bug Investigation]
**Describe the bug**
Jamulus server and client are running on the same Windows machine.
**Situation 1:**
Start Jamulus server (leave the app maximized), start Jamulus client, connect to server.
Everything fine!

**Situation 2:**
Start Jamulus server and minimize it to the task bar, start Jamulus client, connect to the server.
I only hear a kind of “underwater sounds”!

**To Reproduce**
See above. I tried it multiple times, each time the same result

**Expected behavior**
I would expect, both in situation 1 and in situation 2, a working connection with normal sounds.

**Screenshots**

**Operating system**

Windows 11

**Version of Jamulus**

**Comment by pljones** [comment]
Can you confirm the version of Jamulus?

Have you optimised Windows 11 for server usage?  i.e. prioritise background tasks, etc.

What else have you got running on Windows?

Have you tried contacting Microsoft Support to ask why this happens?

**Comment by Willem119** [comment]
It is Jamulus 3.8.1.
No, I don’t have optimised Windows 11.   However I used this setup for 2 years already and till now I had never issues with it (we are a group of 15 people and were always using Jamulus in this setup).
Furthermore no additional applications running.
Why should I call Microsoft? I don’t have any other issues with the configuration.

**Comment by pljones** [comment]
You've been using Windows 11 for two years?  Incredible.  You must be on the developers group.

**Comment by Willem119** [comment]
And so….. it is not an issue??

**Comment by pljones** [comment]
I haven't said that but it would be useful to get a clearer understanding of your set up.

Given that nothing has changed in your set up at all, according to you, can you think of anything that can be causing this issue now that wasn't before?

**Comment by Willem119** [comment]
We were using Jamulus a lot of times (more than twice a week) between november 2020 and june 2021.
No serious issues then!
Next a couple of months nothing.
A few weeks ago we started again using Jamulus.
Major updates:
-  I upgraded from Jamulus 3.6.2 to Jamulus 3.8.1
-  I (indeed) upgraded from Windows 10 to Windows 11
- I loaded the latest audio drivers for my Audient Evo 8 audio interface

I noticed that sometimes everyting was ok (sound was ok), but frequently we had this “underwater noises”.
Today I discovered that the “underwater noises” were only popping up in the situation where I minimized the server app before starting the client app.

**Comment by ghost** [comment]
This is definitely nothing other than a recent MSFT regression.
See #1855
If Windows can be optimised for server usage, maybe that can help.

**Comment by Willem119** [comment]
I read the complete 1855 topic, but I don’t see  why it is similar to my topic about minimizing/maximizing the server app to the task bar.

**Comment by pljones** [comment]
Folding at home appears to use background processing time on a PC for distributed computing.  I would imagine installing that would optimize your computer pretty well for background tasks, which is what real time multithreaded processing software like Jamulus needs.

Upgrading from Windows 10 to Windows 11 is likely to have removed any such optimisations you might have made in the past, meaning your computer is no longer fit for purpose - it's fit for what Microsoft think you want to do.

Hence, it's a Microsoft problem.  You'll need to get guidance on optimising Windows 11 for background processing and follow it.

**Comment by henkdegroot** [comment]
Just want to add, I could not reproduce this on my Windows 10 system with the latest Jamulus version.
Also in Windows 11 Microsoft has worked hard to ensure that CPU and memory resources are always available and ready to go for foreground tasks. When you minimize the Jamulus Server application, it changes from a foreground task into a background service, hence why you most likely see this side effect.

Perhaps you can try to start the Jamulus Server app using a higher priority and see if that resolves the issue?

From the command prompt you can do that using this command:

`start "" /High "C:\Program Files\Jamulus\Jamulus.exe" -s`

There are other options to do this, but I believe this is the easiest to test.

_EDIT:_ I have now been able to test with a Windows 11 setup and can report to "hear" the same. It does not matter when I minize the Jamulus Server. I can start the app normal and get the client connected. All sounds good and normal. When I then mimize the Jamulus Server app, after a few seconds the sounds starts "gurgling". Sounds really like a Win 11 issue to me. There is no problem when I run the Jamulus Server from the command line with the "nogui" option. Not sure if you need the GUI or not. If you do need the GUI, I would just recommend to move the Jamulus Server window almost off screen.

And to add, the same laptop with Windows 10 installed does not have this effect.
One difference that I do note is that on Windows 10, the Jamulus Server app is indeed moving to the "background" processes list (in task manager) when minimized, while in Windows 11 the app remains in the top section (applications). Windows seems to be dealing different with the minimize option.

**Comment by Willem119** [comment]
Yes, thanks for the input and suggestions!
I also did a few additional tests (with the same results).
Knowing now that, when leaving the server application maximized, everything works fine it is not a big issue for me anymore.

**Comment by ann0see** [comment]
Probably worth requesting high priority programmatically. It should be somehow possible (e.g similar to https://forum.qt.io/topic/115112/how-to-set-application-execution-priority-to-high/6  or https://social.msdn.microsoft.com/forums/windowsdesktop/en-US/92d57992-25fb-45f6-bcb3-4fd2ab9d995a/how-can-i-prioritise-io-for-my-application-at-the-expense-of-rest-of-system)

**Comment by henkdegroot** [comment]
> high priority

This is already in place and does not seem to have an effect. Even when I manually set it to Realtime, the issue still occurs.

Will need to perform further investigation on this.

**Reply by henkdegroot** [reply]
Just an update, so indeed the code in Jamulus is setting the application priority to High. Which can been seen in the windows task manager as well, so that works. No difference spotted between Windows 10 and Windows 11 here.
Another behaviour I found is that when the Jamulus Server dialog is visible, but I then open another application (like Notepad) and maximize the window, the "sound" in Jamulus is getting problems.
On my Windows 11 system I can see the CPU usage is lowering when these sound problems occur. As an example, I run just Jamulus Server with 1 connected client and CPU for the Jamulus Server app is like at max. 3%. When I minimize the Jamulus Server dialog, the CPU drops down to 0.4% or so. This is a true difference between Win 10 and Win 11. Doing the same thing on Win 10 does not change the CPU usage.
Now need to figure why Windows is doing this (obvious they claim it is to improve applications which have "focus") and then how to prevent this from happening. I will be doing some more experiments on this.

Any thoughts on this, then please share.

**Reply by hoffie** [reply]
I'm wondering whether this blog post about power management might lead to some new ideas:
https://www.anthropicstudios.com/2022/01/13/asking-windows-nicely/#trick-1-asking-windows-nicely

**Reply by henkdegroot** [reply]
@hoffie, nice to see a comment from you and thank you so much for the hint. Found some useful detail in this and more important it seems to work as well.

**Reply by henkdegroot** [reply]
Unfortunately this trick is not resolving the issue. I only thought it did, because I was launching jamulus -s from a command prompt with admin rights in Windows. Launching jamulus server with admin rights also fixes this issue.

**Comment by gilgongo** [comment]
@Willem119 Hi - I've moved this to a discussion so that we can agree on a spec for any work, and then raise an issue (or preferably a PR) once that's done. Thanks.

**Comment by hoffie** [comment]
There is a small chance that moving our builds to Qt6 might help with this issue.
If someone wants to try, feel free to [download](https://github.com/jamulussoftware/jamulus/suites/5089746973/artifacts/151918119) the completely **untested** Qt6 Windows build from PR #2300. Expect breakage though.

Please post in this discussion, if you can still observe the described bug with this build or if things get better or worse.
Please post to #2300 if you find any other issues with this build.

**Reply by henkdegroot** [reply]
@hoffie, I had the same thought....and tested....but unfortunately no difference here.

**Reply by hoffie** [reply]
Thanks for testing!
Too bad :(
Can you confirm that the build is correct, i.e. that it shows Qt6 in the About Qt screen?

**Reply by ann0see** [reply]
Tested on my Windows 11 machine and it showed QT6.
